File: src/main/java/com/vielheit/graph/service/EntryService.java
package com.vielheit.graph.service;

import com.vielheit.graph.domain.Entry;
import com.vielheit.core.service.Service;

import javax.transaction.Transactional;
import java.util.List;
import java.util.Optional;

@Transactional
public interface EntryService extends Service {
    Optional<Entry> saveEntry(Entry entry);

    Optional<List<Entry>> getEntries(Long userId);
}

File: src/main/kotlin/cc/kojeve/vielheit/service/EntryService.kt
package cc.kojeve.vielheit.service

import cc.kojeve.vielheit.domain.Entry
import cc.kojeve.vielheit.dto.EntryData
import cc.kojeve.vielheit.repository.UserRepository
import cc.kojeve.vielheit.request.EntryRequest
import cc.kojeve.vielheit.util.RestException
import org.springframework.data.repository.CrudRepository

class EntryService(
        override val repository: CrudRepository<Entry, Long>,
        val userRepository: UserRepository
) : Service<Entry, EntryData, EntryRequest>() {

    override fun save(req: EntryRequest): EntryData {
        val user = userRepository.findById(req.userId).orElseThrow { RestException.BadRequest() }

        val entry = Entry(user, req.title)
        repository.save(entry)

        return EntryData(entry)
    }
}